src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -347,19 +345,24 @@
    [:ShutdownService shutdown-on-error]]
 
   (start [this context]
-         (start-puppetdb context (get-config) this add-ring-handler get-route shutdown-on-error))
-
-  (stop [this context]
-        (stop-puppetdb context false))
-  (shared-globals [this]
-                  (:shared-globals (service-context this)))
+         (let [config (-> (get-config)
+                          conf/process-config!)
+               url-prefix (get-route this)
+               shutdown-on-error* (partial shutdown-on-error (service-id this))
+               add-ring-handler* #(->> %
+                                       (compojure/context url-prefix [])
+                                       (add-ring-handler this))]
```

Comment:
The idea here was separating out anything that needs `this` from the start-puppetdb function to make it a little less monolithic and see what kinds of trapperkeeper related work we're doing from the defservice.